src/sage/rings/abc.pyx

```diff
@@ -66,13 +66,13 @@ class UniversalCyclotomicField(Field):
     EXAMPLES::
 
         sage: import sage.rings.abc
-        sage: K = UniversalCyclotomicField()                                            # needs sage.rings.number_field
-        sage: isinstance(K, sage.rings.abc.UniversalCyclotomicField)                    # needs sage.rings.number_field
+        sage: K = UniversalCyclotomicField()                                            # needs sage.libs.gap
+        sage: isinstance(K, sage.rings.abc.UniversalCyclotomicField)                    # needs sage.libs.gap
```

Comment:
Perhaps I am repeating myself. I don't like adding "needs" tags that is dependent on implementation details of a (mathematical) feature.

In this case, `CyclotomicField` needs gap but this is implementation detail.

"# needs sage.rings.number_field" is good. Users of cyclotomic fields would understand "sage.rings.number_field" is needed. But most of them (me included) won't understand the need of gap. This implementation detail may change.

Lots of similar changes are made below.

Why this "regression"?

Reply:
I also didn't know that the UCF is implemented using GAP before I built and tested the distributions that provide these features. Our implementation of number fields (including the specific cyclotomic fields) already involves all of PARI, NTL, FLINT, and the LinBox libraries. The UCF adds GAP to that and because this is a big system, I want to distinguish it. I'll be happy to introduce a feature `sage.rings.universal_cyclotomic_field` and use it instead of sage.libs.gap. How does this sound?

Reply:
Here `# needs sage.rings.number_field` is replaced with `# needs sage.libs.gap`. Would this work?

If UCF needs gap on top of number fields, shouldn't it be `# needs sage.rings.number_field  sage.libs.gap`?

Perhaps it is not economic to create the new feature `sage.rings.universal_cyclotomic_field`, If it is (probably) only UCF that needs feature `sage.rings.universal_cyclotomic_field`.

Or why not just add GAP to the set of libraries  PARI, NTL, FLINT, and the LinBox if it is needed to create universal cyclotomic fields? Would this (more libraries) do any harm to users of number fields?

Reply:
> If UCF needs gap on top of number fields, shouldn't it be `# needs sage.rings.number_field sage.libs.gap`?

Yes, that would make sense and is suitable as an idiom. I'll make this change.

Reply:
I guess features like `sage.rings.number_field` already cut much of unrelated sage library and libraries. I think we should be generous in including libraries to provide a feature, and not try to achieve the minimum in the number of included libraries.

Reply:
GAP is huge; in our standard installation it takes half a gigabyte. Not installing it, unless it is needed, is very much the point of the modularization...
```
$ du -sh local/lib/gap
534M	local/lib/gap
```

Reply:
I see the point. Everything depends on how often (which is subjective) the component (here GAP) is used in a feature. If GAP is only used for UCF and UCF is infrequently used by the users of number fields, your decision is reasonable.

OK for now. Thanks for the changes.